src/main/java/nl/rijksoverheid/mev/gezagsmodule/service/newversion/GezagServiceNew.java

```diff
@@ -80,31 +80,32 @@ public GezagAfleidingsResultaat getGezagAfleidingsResultaat(final String bsn, fi
         GezagAfleidingsResultaat result;
         List<Gezagsrelatie> gezagRelaties = new ArrayList<>();
         String route;
-        Persoonslijst plPersoon = null;
+        Optional<Persoonslijst> plPersoon = Optional.empty();
         GezagBepaling gezagBepaling = null;
         try {
             if (new BSNValidator().isValid(bsn)) {
                 plPersoon = brpService.getPersoonslijst(bsn, transaction);
-                transactionHandler.saveGezagmoduleTransaction(
-                    PersoonlijstType.PERSOON,
-                    plPersoon.getReceivedId(),
-                    null, null, null, transaction);
+                if (plPersoon.isPresent()) {
+                    Persoonslijst persoon = plPersoon.get();
```

Comment:
`Optional#isPresent` + `Optional#get` can be replaced with `Optional#ifPresent`

Reply:
Never mind. Variables inside lambdas need to be effectively final and too many mutations are taking place.